Update printer.cfg

akinferno opened the pull request:
Corrected [fan_generic fan0] pin definition. Pin PB0 is from the SV08 toolhead board for fan3. The Zero toolhead board uses PA7.
Added description of the fan in the comment.

akinferno committed:
Update printer.cfg

Corrected [fan_generic fan0] pin definition.
Added description of the fan in the comment.

DivineGod commented:
Did you confirm this with the printer running?

I think the PDF showing the toolhead board pin-out is wrong. Since Fan2 and Fan3 are showing PA7 as the connected pin. However, PB0 does control the Part Cooling Fan correctly on my specific Sovol Zero unit.

akinferno commented:
> Did you confirm this with the printer running?
> 
> I think the PDF showing the toolhead board pin-out is wrong. Since Fan2 and Fan3 are showing PA7 as the connected pin. However, PB0 does control the Part Cooling Fan correctly on my specific Sovol Zero unit.

PA7 is correct on my printer, and my friends. The machines were ordered at different times, his from Sovol and mine from Fabreeko. Nothing is plugged into the 'model fan' from the PDF, but the fan next to the hotend or 'cooling fan' or FAN2 on the PCB. Maybe FAN3 or 'Model Fan' is PB0?  But the clearly one is mislabeled. Ours were sticky glued into the top FAN2 port. To complicate things, one response on my Facebook post also states that their fan works correctly on PB0. 

The model on my board is MINI_ZJB_V1.1.  The manufacture date is 20241204.

How I noticed: My friend lives an hour away, so I drove out to help him with a couple printers. He kept getting failed prints, when the printer had been working great for weeks. He prints to sell, so he kept the printing working. I noticed his fan wasn't spinning and I couldn't turn it on through console, Mainsail or the display. So I removed it, assuming it was dead, and brought it home to test in my machine, to verify it wasn't the toolhead board. So he would know what to have Sovol replace. I first verified my machine fan was working, and it was not. I primarily print small ABS parts, so hadn't had noticed my part fan wasn't working. I changed the config to PA7, based on the diagram, and [fan0] powered right up. My friend picked up his fan from me, and reinstalled it. I remoted into his machine, updated his config to use PA7 and when I powered on the fan in Mainsail, he said it was spinning. No more print failures. Our fans are plugged into the port that is printed as Fan2 on the board. Right next to the 'cooling fan' or hotend fan.

I know this is long. But the diagram needs to be corrected to show the proper pinouts. FAN1 is PA6, FAN2 is PA7, maybe FAN3 is PB0 (maybe?). If so, my friend and my printers were assembled incorrectly. If the part fan is supposed to be in FAN3, since it is labeled 'model fan' on the diagram. I will correct the wiring and restore the config so it won't break with future updates.

akinferno commented:
Okay, upon verification, the FAN3 port is PB0. So, the printer.cfg isn't wrong, our wiring was. I will close the pull request, but the PDF needs to updated.
